tests/docs_test.go

```diff
@@ -0,0 +1,71 @@
+// +build system
+// +build docs
+
+package tests
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/mccanne/zq/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var doctests = []struct {
+	name string
+	cmd  string
+
+	expected string
+}{
+	{
```

Comment:
Also, if it's more useful for any upstream tooling, we could have this be a json file such as `doctests.json` with the same contents in a json array.

Reply:
That's certainly interesting to think about. When I started this exercise I was wondering if it would help to end up with a way to add tests without having to write any Go. It looks like the "formats" tests are precisely this, but they're also ultra-simple in that they seem to be all-in/all-out without any reliance on ZQL or `zq` flags. Something like you've proposed where it all goes in a JSON array feels like it could achieve the "no Go" aim, though maybe it wouldn't be seen as such a big win since anyone else contributing to the project is likely to be a Go programmer and hence isn't going to perceive the language as a learning curve like I do... or maybe there'd still be value in presenting the testing framework as being that simple as opposed to expecting them to copy/modify existing Go code from other tests?

That said, if all done in JSON, now one would have to do annoying escapes, and having just seen a way to avoid them by using the backticks, I'm conflicted. 😄

Reply:
@philrz I guess part of this depends on what we define as "no Go" :). The present shape does technically require editing a go file, but all that editing can stay within the test list, which can be done while remaining blissfully non-Go-aware (ok, other than the escapes). 

The aspect I was thinking about was really for tooling/automation. For example if the test list was in `.js` would be easy to write a simple shell scripts to add a new query to the list, compare the list of test queries with a list of doc queries, or even run the set of tests and comparing the results to expected during `diff`. Your call as to if/when anything like this is of use.